src/useResizeDetector.ts

```diff
@@ -36,7 +38,7 @@ function useResizeDetector<RefType extends Element = Element>(props: FunctionPro
     height: undefined
   });
 
-  useLayoutEffect(() => {
+  useEnhancedEffect(() => {
     if (isSSR()) {
       return;
```

Comment:
Just as a side note: If the hook would actually do something which depends on targetRef these changes would be problematic in some cases (because of `const ref: MutableRefObject<null | RefType> = targetRef ?? localRef;`)